### PR TITLE
adding statistical tests of sweeps (WIP)

### PR DESCRIPTION
**Pull request by andrewkern**
I added an example statistical test between the msprime implementation and my old discoal implementation. I'm not 100% confident that all the scaling conversions are good to go between discoal and msprime however. @petrelharp looked over my should a bit today and we got it pretty close but I'm not sure how to scale the selection coefficient between discoal and msprime. 

from playing with it way too much it looks like the \alpha_discoal * 2 = \alpha_msprime, but I'm scratching my head as to why.

### CODE REVIEW

File: verification.py [hunk 1]
```diff
@@ -363,6 +363,58 @@ def f():
...
+        seqlen = 1e4
+        mu = 2.5e-4
+        rho = 2.5e-4
+        mod.position = np.floor(seqlen/2)
+        data = collections.defaultdict(list)
+        replicates = msprime.simulate(10,  model=mod, length=seqlen,
+                                      num_labels=2, recombination_rate=rho,
+                                      num_replicates=nreps)
+        for x in replicates:
+            ts = msprime.simulate(from_ts=x, recombination_rate=rho)
```
Comment (andrewkern) [inline verification.py]: this is a bit ugly, but I'm forced to do this as our current implementation of the `SweepGenicSelection()` model doesn't finish all the coalescing for us. so what i've done here is start a second round of simulation off from the tree sequence that the sweep simulation ends with. 

@jeromekelleher what do you think here?

File: verification.py [hunk 2]
```diff
@@ -363,6 +363,58 @@ def f():
...
+            data["D"].append(tsm.Tajimas_D())
+            data["ss"].append(tsm.segregating_sites(span_normalise=False))
+        data["pi"] = np.array(data["pi"]).flatten()
+        data["D"] = np.array(data["D"]).flatten()
+        data["ss"] = np.array(data["ss"]).flatten()
+        df = pd.DataFrame.from_dict(data)
+        df = df.fillna(0)
+        df_d = self._run_discoal_mutation_stats(args)
+        df_df = df_d[["pi", "D", "ss"]]
+        print("msp pi mean: ", df['pi'].mean())
```
Comment (andrewkern) [inline verification.py]: this is just sanity checking stuff that is helping me debug scaling. I imagine I'll take it out before we merge

File: lib/dev-tools/example.cfg [hunk 3]
```diff
@@ -32,17 +32,17 @@ model = {
     #truncation_point = 10.0;
 
     # The discrete time Wright Fisher
-    /* name = "dtwf"; */
```
Comment (andrewkern) [inline lib/dev-tools/example.cfg]: not sure what happened to these comments on master... shouldn't they be #'s?

File: lib/msprime.c [hunk 4]
```diff
@@ -6284,8 +6292,8 @@ msp_set_simulation_model_sweep_genic_selection(msp_t *self, double reference_siz
         ret = MSP_ERR_BAD_SWEEP_POSITION;
         goto out;
     }
-    if (start_frequency < 0.0 || start_frequency > 1.0
-            || end_frequency < 0.0 || end_frequency > 1.0) {
+    if (start_frequency <= 0.0 || start_frequency >= 1.0
```
Comment (andrewkern) [inline lib/msprime.c]: need to change these to avoid weird behavior in trajectory simulation

File: verification.py (hunk 1 above)
Comment (jeromekelleher) [reply verification.py]: How about adding  a ``demographic_events=SimulationModelChange(time=None, model="hudson")`` ? That was the idea anyway, I'm not sure if I got around to implementing this. We should change the default value for ``time`` to None if this works, so we can just say

```
ts = msprime.simulate(
      10,  
      model=selection_model, 
      demographic_events=[SimulationModelChange(model="hudson")])
```
and it'll do what we want.

Comment (andrewkern) [reply verification.py]: Yeah it doesn't look like a `time=None` option has been implemented for this situation where you want it to change models at some arbitrary time.

Comment (andrewkern) [reply verification.py]: seems to me like setting `time=np.Inf` is doing the correct thing but i'm not 100% certain

**Comment by jeromekelleher** [comment]
@andrewkern, can we update this and get it merged soon please? We can register issues to follow up on, but it'd be nice to get this merged (we want to do some reformatting of the code using Black and clang-format, so any open PRs will be disrupted)

**Comment by jeromekelleher** [comment]
@andrewkern, looks like the actual changes here are small. Mind if I clean this up a bit and merge it? We can pick up the actual testing stuff again afterwards.

**Comment by andrewkern** [comment]
@jeromekelleher sorry been missing these notifications! yes merge away!!

pinging @daniel-goldstein here as the verification params I am using should be appropriate

**Comment by jeromekelleher** [comment]
I've tidied this up @andrewkern and opened up a few issues (#941, #942, #943) to follow up on. I changed few things back to the way they were, most notably the default params for the sweep model. I think it's a Bad Idea to put in arbitrary values here and users are much better off being forced to decide what they should be.

**Pull request merged**